### PR TITLE
add inference config for raw images in detection module, update readme

### LINKED ISSUE #166: Lung Nodule detection (Inference)
I have followed the instructions to do the inference but I have three questions: 
1- The output I got having three main things: 
- the ground truth coordinates
- label 
- image name
I’m wondering why the label is always zero? but some images have two zeros and some have more? can you please elaborate more on this part?

2- The inference is working only on the raw data (The unprocessed images), while when I infer the model on the resampled data it is not working!?

3- The inference uses the ground truth and calculates the label score, what if I want to do the inference on a different dataset that has no ground truth? 
I want to see the performance of the model on another dataset.

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes #166 .

### Description
Add inference config file for raw input images.
Update readme to add more explanation on inference

### Status
**Ready/Work in progress/Hold**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @Can-Zhao ,

To avoid duplicated config content, I would suggest to just define `inference_raw.json` to override `inference.json`. Like your `evaluate.json` to override the `train.json`:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/configs/evaluate.json
Or maybe you can just add a flag in the `inference.json` to control the raw or not-raw, refer to:
https://github.com/Project-MONAI/model-zoo/pull/163
CC @yiheng-wang-nv 

Thanks.

**Comment by Can-Zhao** [comment]
> Hi @Can-Zhao ,
> 
> To avoid duplicated config content, I would suggest to just define `inference_raw.json` to override `inference.json`. Like your `evaluate.json` to override the `train.json`: https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/configs/evaluate.json Or maybe you can just add a flag in the `inference.json` to control the raw or not-raw, refer to: #163 CC @yiheng-wang-nv
> 
> Thanks.

Thank you! I make it override the inference.json

**Review by yiheng-wang-nv — COMMENTED** [review]
Hi @Can-Zhao , please also modify the `version` and `changelog`, thanks!

File: models/lung_nodule_ct_detection/configs/inference_raw.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+    "data_list_file_path": "$@bundle_root + '/LUNA16_datasplit/original/dataset_fold0.json'",
+    "data_file_base_dir": "/home/canz/Projects/datasets/LUNA16/93176/Images",
+    "preprocessing": {
+        "_target_": "Compose",
+        "transforms": [
+            {
+                "_target_": "DeleteItemsd",
```
Comment (Nic-Ma) [inline models/lung_nodule_ct_detection/configs/inference_raw.json]: May I know the error or problem if don't remove these 2 keys?

Thanks.

File: models/lung_nodule_ct_detection/configs/inference_raw.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                "keys": [
+                    "box",
+                    "label"
+                ]
+            },
+            {
+                "_target_": "LoadImaged",
+                "keys": "image",
+                "meta_key_postfix": "meta_dict",
+                "reader": "itkreader",
```
Comment (Nic-Ma) [inline models/lung_nodule_ct_detection/configs/inference_raw.json]: I feel the only difference in this new config is this line, no need to override all the transforms.
Maybe you can just add a flag in the original `inference.json`? For example:
```json
"resampled_image": "true",
...
"reader": "$'nibabelreader' if @resampled_image else 'itkreader'",
...
```

What do you think?
CC @yiheng-wang-nv @wyli .

Thanks.

Comment (yiheng-wang-nv) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: Agree. Similar for the `Spacingd` transform, can add a flag and make use of `_disabled_` to control whether to use this transform. Can refer to [here](https://github.com/Project-MONAI/model-zoo/pull/163/files#diff-53004b9136c53f2fc7834c750c784abc55fc3161cfe1c53c1143dde5ae581a23R127)

File: models/lung_nodule_ct_detection/configs/inference_raw.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+    "data_list_file_path": "$@bundle_root + '/LUNA16_datasplit/original/dataset_fold0.json'",
```
Comment (Can-Zhao) [inline models/lung_nodule_ct_detection/configs/inference_raw.json]: If we use `"_disabled_"`, how to handle the change of `"data_list_file_path"` of the two setting? @Nic-Ma @yiheng-wang-nv . Sorry, I'm still not that familiar with the commands of bundle

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 1 above)
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: There is no error resulted. I have removed this transform. I added only to make sure the inference did not use ground truth.

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 2 above)
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: There are at least three places to switch. I was afraid it is not that convenient to use `"_disabled_"`

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 3 above)
Comment (Nic-Ma) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: Actually, I feel we don't need to care about the `data_list_file_path`, if it's raw image, you need to set `resampled_image=True`.
Don't worry about the bundle APIs, I think @yiheng-wang-nv can help you on the development.

Thanks.

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 2 above)
Comment (Nic-Ma) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: Hi @yiheng-wang-nv , I don't see any difference in the `Spacingd` transform, what do you mean to change for `resampled or raw`?

Thanks.

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 3 above)
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: But here, we provided download link for all the data list json files. I put json files for resampled and raw images in two different folders in the downloaded file. It is already structured. That's why I put a `"data_list_file_path"` here.

File: models/lung_nodule_ct_detection/configs/inference_raw.json (hunk 2 above)
Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference_raw.json]: There is a change. When apply to resampled image, we should not include spacing

File: models/lung_nodule_ct_detection/configs/inference.json [hunk 4]
```diff
@@ -71,16 +73,18 @@
...
             {
-                "_target_": "DeleteItemsd",
-                "keys": [
-                    "box",
-                    "label"
-                ]
+                "_target_": "LoadImaged",
+                "keys": "image",
+                "meta_key_postfix": "meta_dict",
+                "_disabled_": "@whether_raw_luna16"
```
Comment (yiheng-wang-nv) [inline models/lung_nodule_ct_detection/configs/inference.json:79]: Hi @Can-Zhao , thanks for the updates. Here I think we may not need two `LoadImaged `, and you can use:
```
            {
                "_target_": "LoadImaged",
                "keys": "image",
                "meta_key_postfix": "meta_dict",
                "reader": "$'itkreader' if @whether_raw_luna16 is True else None",
                "affine_lps_to_ras": "@whether_raw_luna16",
            },
```
In addition, according to the readme file, should the `affine_lps_to_ras` of `AffineBoxToWorldCoordinated` be consistent with this transform?

Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference.json:79]: I'm afraid they cannot be combined.
1) the resampled images are nifti. nifti reader does not take `affine_lps_to_ras` as an argument, it will give an error.
2) that's why I put two `"LoadImaged"` here.

Comment (yiheng-wang-nv) [reply models/lung_nodule_ct_detection/configs/inference.json:79]: Thanks @Can-Zhao , I double checked the settings with nii format data and you are right.
Could you please also update the changelog and version in metadata as I mentioned in a previous comment? It is a necessary step before merging a PR, thanks again!

Comment (Can-Zhao) [reply models/lung_nodule_ct_detection/configs/inference.json:79]: Thank you! Just changed them!

**Review by yiheng-wang-nv — APPROVED** [review]
